server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java

```diff
@@ -487,6 +487,13 @@ public void testExplain() throws Exception {
             )
             .setRefreshPolicy(IMMEDIATE)
             .get();
+
+        // refresh is important to enforce creation of a segment for document ingest above
+        // and prevent merging with deleted segemnts created by the index random function
+        // this ensures that the range doesnt change for the child docs asserted below
+        // ref: https://github.com/opensearch-project/OpenSearch/issues/11413
+        refresh("test");
```

Comment:
@neetikasinghal I am confused by this change to be fair, the statement just above adds the single document with `RefreshPolicy.IMMEDIATE`, that is:

```
        /**
         * Force a refresh as part of this request. This refresh policy does not scale for high indexing or search throughput but is useful
         * to present a consistent view to for indices with very low traffic. And it is wonderful for tests!
         */
        IMMEDIATE("true"),

```

I suspect that either a) `RefreshPolicy.IMMEDIATE` is not respected, clearly a bug or b) this is noop. Could you please clarify my understanding here? Thank you.

Reply:
In addition to @reta's comment above, can't merges still happen as a part of `indexRandomForConcurrentSearch("test")`? I don't see how this would prevent merges

Reply:
@reta thanks, you are right, there is already a refresh being called as part of the request above. Let me double check on why this solves the issue or try a few more runs to validate if there is any failure.

Reply:
this was working in my local due to random seed getting messed up in IntelliJ, however this is not the right solution.